ui/src/containers/NodeVolumes.js

```diff
@@ -31,44 +42,109 @@ const VolumeTable = styled.div`
   margin-top: ${padding.small};
 `;
 
+const DeleteConfirmationModal = {
+  content: {
+    top: '50%',
+    left: '50%',
+    width: '540px',
+    height: '190px',
+    transform: 'translate(-50%, -50%)'
+  }
+};
+
+const NotificationButtonGroup = styled.div`
+  display: flex;
+  justify-content: center;
+  margin-top: ${padding.larger};
+  margin-left: 270px;
+`;
+
+const DeleteButton = styled(Button)`
+  margin-left: ${padding.small};
+`;
+
 const NodeVolumes = props => {
+  const { intl } = props;
+
   useRefreshEffect(refreshVolumesAction, stopRefreshVolumesAction);
   useRefreshEffect(
     refreshPersistentVolumesAction,
     stopRefreshPersistentVolumesAction
   );
+
   const volumes = useSelector(state => state.app.volumes);
+  const persistentVolumes = useSelector(state => state.app.volumes.pVList);
   const [sortBy, setSortBy] = useState('name');
   const [sortDirection, setSortDirection] = useState('ASC');
+  const [
+    isDeleteConfirmationModalOpen,
+    setisDeleteConfirmationModalOpen
+  ] = useState(false);
+  const [deleteVolumeName, setDeleteVolumeName] = useState('');
   const onSort = ({ sortBy, sortDirection }) => {
     setSortBy(sortBy);
     setSortDirection(sortDirection);
   };
 
+  const isVolumeDeletable = rowData => {
+    const volumeStatus = rowData.status;
+    const volumeName = rowData.name;
+
+    if (
+      volumeStatus === STATUS_VOLUME_UNKNOWN ||
+      volumeStatus === STATUS_PENDING ||
+      volumeStatus === STATUS_TERMINATING
+    ) {
+      return false;
+    } else if (
+      volumeStatus === STATUS_FAILED ||
+      volumeStatus === STATUS_AVAILABLE
+    ) {
+      if (persistentVolumes === []) {
+        return true;
+      } else {
+        const persistentVolume = persistentVolumes.find(
+          pv => pv?.metadata?.name === volumeName
+        );
+        const persistentVolumeStatus = persistentVolume?.status?.phase;
+        if (
+          persistentVolumeStatus === STATUS_BOUND ||
```

Comment:
For `BOUND` you want to return `false` (a PV Bound is a PV in use => cannot delete).